src/config.ts

```diff
@@ -488,3 +499,24 @@ export function mergeStyle(
 export function stringOrNull(spec: unknown): string | null {
   return spec == null || spec === false ? null : String(spec);
 }
+
+function normalizeDuckDB(spec: unknown): DuckDBConfig {
+  const install = spec?.["install"] ?? ["json", "parquet"];
+  const load = spec?.["load"] ?? [];
+  const source = new Map(Object.entries(spec?.["source"] ?? {}));
+  return {
+    bundles: ["eh", "mvp"],
+    install,
+    load: load.filter((name: string) => install.includes(name)),
+    source: Object.fromEntries(
+      install.map((name: string) => {
+        let href = source.get(name) ?? "core";
+        if (href === "core") href = "https://extensions.duckdb.org";
+        else if (href === "community") href = "https://community-extensions.duckdb.org";
```

Comment:
If we want to also allow registration of (potentially non-self-hosted) extensions at runtime, this will need to move to src/client/stdlib/duckdb.js rather than here… it might also be more appropriate to move this logic to a function exported by src/duckdb.ts so that it’s more centralized.

Like what if you could say:

```js
DuckDBClient.of(…, {
  extensions: {
    spatial: {
      load: true,
      source: import.meta.resolve("duckdb:core:spatial")
    }
  }
})
```

Potentially with shorthand (we could know that `spatial` isn’t one of the automatically loaded extensions and load it by default):

```js
DuckDBClient.of(…, {extensions: {spatial: import.meta.resolve("duckdb:core:spatial")}})
```

Because this uses `import.meta.resolve` this would mean that the spatial extension would be self-hosted the same way as if you had used the **duckdb** config option (or front matter), i.e., it would come from /_duckdb/.

Reply:
I’m not sure if `import.meta.resolve("duckdb:core:spatial")` should resolve to the full path to the extension…

```
/_duckdb/extensions.duckdb.org/v1.1.1/wasm_eh/spatial.duckdb_extension
```

Or if it should resolve to the self-hosted [“custom repository”](https://duckdb.org/docs/extensions/working_with_extensions#creating-a-custom-repository) folder:

```
/_duckdb/extensions.duckdb.org/
```

Since the latter is the path that you’d need to pass to `INSTALL spatial FROM '<repository>'`; The former is only possible if we only support the `eh` platform. So, I think I’m inclined to do the latter. So `import.meta.resolve("duckdb:core:spatial")` would download the needed files for that extension from extensions.duckdb.org, but the resulting value would be just the custom repository location, meaning that e.g. `import.meta.resolve("duckdb:core:parquet")` would resolve to the same value (even though it would download different files).